### PR TITLE
Uml 1061 fix cancelled lpa view

### PR DESCRIPTION
**Pull request by williamfalconeruk**
# Purpose

To fix issue where an LPA is cancelled and the share code is not, you still need to see the cancelled LPA.

Fixes UML-1061

## Approach

- fix flow in handler
- added logging for cancelled codes (refactor)
- update Sirius object definition to include cancellation date
- rework behat tests
- additional fix to api client to get the edge case for messages in test mocks to work (will need revisiting)

TODO:
- [x] add test for cancellation date display (as we need to mock that out too)
- [x] check flow for 'Revoked' (unsure on this one)
- [x] Update Translation Files and send to lewis
- [x] add unit tests to up patch coverage

## Learning
N/A

## Checklist

* [x] I have performed a self-review of my own code
* [x] I have added relevant logging with appropriate levels to my code
* [ ] I have updated documentation (Confluence/GitHub wiki/tech debt doc) where relevant
* [x] I have added tests to prove my work
* [ ] The product team have tested these changes

### CODE REVIEW

**Comment by williamfalconeruk** [comment]
This is just waiting for pot / po file update.

File: service-front/app/test/CommonTest/Service/Lpa/LpaServiceTest.php [hunk 1]
```diff
@@ -157,6 +157,30 @@ public function it_gets_an_lpa_by_passcode_and_surname_for_full()
         $this->assertInstanceOf(Lpa::class, $lpa->lpa);
     }
 
+
+    /** @test */
+    public function it_finds_a_cancelled_lpa_by_passcode_and_surname()
```
Comment (williamfalconeruk) [inline service-front/app/test/CommonTest/Service/Lpa/LpaServiceTest.php:162]: Rename this

File: service-front/app/test/CommonTest/Service/Lpa/LpaServiceTest.php [hunk 2]
```diff
@@ -157,6 +157,30 @@ public function it_gets_an_lpa_by_passcode_and_surname_for_full()
...
+
+        $this->expectException(ApiException::class);
+        $this->expectExceptionCode(StatusCodeInterface::STATUS_GONE);
+        $this->expectExceptionMessage('Share code cancelled');
+
+
+        $service->getLpaByCode('P9H8-A6ML-D3AM', 'Sanderson', false);
+    }
     /** @test */
     public function it_finds_an_expired_lpa_by_passcode_and_surname()
```
Comment (williamfalconeruk) [inline service-front/app/test/CommonTest/Service/Lpa/LpaServiceTest.php:185]: Rename this too

File: service-front/app/test/CommonTest/Service/Lpa/LpaServiceTest.php [hunk 3]
```diff
@@ -309,6 +334,63 @@ public function it_finds_an_lpa_by_passcode()
         $this->assertEquals($dob, ($lpa['lpa'])->getDonor()->getDob()->format('Y-m-d'));
     }
 
+
+    /** @test */
+    public function it_finds_an_lpa_by_passcode_cancelled()
```
Comment (williamfalconeruk) [inline service-front/app/test/CommonTest/Service/Lpa/LpaServiceTest.php:339]: Rename this

**Review by nickdavis2001 — APPROVED** [review]

**Review by SeemaMenon — COMMENTED** [review]
LGTM.
Though looks merged with master, the front routes file doesn't look up to date with master.

Also getting an error page [Twig invalid extension exception] locally when trying to bring up the service being on the branch.

One more reviewer please

**Review by SeemaMenon — APPROVED** [review]
